src/store/index.ts

```diff
@@ -0,0 +1,87 @@
+import Vue from 'vue';
+import Vuex, { Store } from 'vuex';
+import { createDirectStore } from 'direct-vuex';
+
+import api from '@/api';
+import { WorkspaceSpec } from '@/types';
+
+Vue.use(Vuex);
+
+interface State {
+  workspaces: string[];
+  currentWorkspace: WorkspaceSpec | null;
+}
+
+const {
+  store,
+  rootActionContext,
+  moduleActionContext,
+  rootGetterContext,
+  moduleGetterContext,
+} = createDirectStore({
+  state: {
+    workspaces: [],
+    currentWorkspace: null,
+  } as State,
+  getters: {
+    nodeTables(state: State): string[] {
+      if (state.currentWorkspace !== null && state.currentWorkspace.nodeTables) {
+        return state.currentWorkspace.nodeTables;
+      }
+      return [];
+    },
+    edgeTables(state: State): string[] {
+      if (state.currentWorkspace !== null && state.currentWorkspace.edgeTables) {
+        return state.currentWorkspace.edgeTables;
+      }
+      return [];
+    },
+    graphs(state: State) {
+      if (state.currentWorkspace !== null && state.currentWorkspace.graphs) {
+        return state.currentWorkspace.graphs;
+      }
+      return [];
+    },
+  },
+  mutations: {
+    setWorkspaces(state, workspaces: string[]) {
+      state.workspaces = workspaces;
+    },
+    setCurrentWorkspace(state, workspace: WorkspaceSpec) {
+      state.currentWorkspace = workspace;
+    },
+  },
+  actions: {
+    async fetchWorkspaces(context) {
+      const { commit } = rootActionContext(context);
+      const workspaces = await api.workspaces();
```

Comment:
Probably need to catch errors here.

Reply:
Should we display these errors to the user? Or just log them? I think I left it this way because if this call fails, the app as a whole won't work either, and so we'd want to see the error.

Reply:
That's a good point. Eventually we'll want to use these failures to signal a general "health" indicator somewhere in the side or top bar of the app. Maybe that would be the time to actually catch errors here (and of course, such a mechanism would probably draw its health data from the store).

So, leave it alone for now. I've opened #70 to track the general idea.